### PR TITLE
Beryllium lnwallet 20220404

### PR DESCRIPTION
**Pull request by eoliveros**
- move list_peers to rebalance_channel.
  - includes renaming template
  - renaming the function
  - redirecting the template used when the function is called.
- new buttons for generating the bitcoin address in new_address,
  - bech32
  - p2sh-segwit
- removed the "," at the end of the result for the transactions in templates/lightning/list_transactions.html
- a new window to display the transactions the we received rather than just pay.
- rebalance_channel page - input boxes are required fields.

### CODE REVIEW

File: ln_wallet_endpoint.py [hunk 1]
```diff
@@ -33,13 +33,16 @@ def lightningd_getinfo():
     rpc = LnRpc()
     return render_template('lightning/lightningd_getinfo.html', info=rpc.get_info())
 
-@ln_wallet.route('/new_address')
+@ln_wallet.route('/new_address/<address_type>', methods=['GET', 'POST'])
```
Comment (djpnewton) [inline ln_wallet_endpoint.py:36]: you should probably remove the address type from the url and instead add inside the `if request.method == 'POST':` area `address_type = request.form.get('address_type')`

File: templates/lightning/index.html [hunk 2]
```diff
@@ -28,7 +28,7 @@ <h2>DIAGNOSTIC</h2>
 <h2>BTC</h2>
 <div class='some-page-wrapper'>
     <div class='row'>
-        <a href="new_address">
+        <a href="/ln_wallet/new_address/bech32">
```
Comment (djpnewton) [inline templates/lightning/index.html:31]: the link should just be `/ln_wallet/new_address` in line with the previous comment

File: templates/lightning/list_transactions.html [hunk 3]
```diff
@@ -20,6 +20,9 @@ <h2>BTC TRANSACTIONS</h2>
 		<td>
 		{% for output in tx.outputs %}
 			<a href="{{ bitcoin_explorer }}tx/{{ tx.hash }}">{{ output.sats }},<a>
+			{% if loop.last %}
```
Comment (djpnewton) [inline templates/lightning/list_transactions.html:23]: why repeat the last output?

**Review by djpnewton — CHANGES_REQUESTED** [review]

File: ln_wallet_endpoint.py (hunk 1 above)
Comment (eoliveros) [reply ln_wallet_endpoint.py:36]: it errors when the address is changed to: "/new_address" and adding "address_type = request.form.get('address_type')"
```
app_1                       | [app_core ERROR] Exception on /ln_wallet/new_address [GET]
app_1                       | Traceback (most recent call last):
app_1                       |   File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 2077, in wsgi_app
app_1                       |     response = self.full_dispatch_request()
app_1                       |   File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1525, in full_dispatch_request
app_1                       |     rv = self.handle_user_exception(e)
app_1                       |   File "/usr/local/lib/python3.9/site-packages/flask_cors/extension.py", line 165, in wrapped_function
app_1                       |     return cors_after_request(app.make_response(f(*args, **kwargs)))
app_1                       |   File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1523, in full_dispatch_request
app_1                       |     rv = self.dispatch_request()
app_1                       |   File "/usr/local/lib/python3.9/site-packages/flask/app.py", line 1509, in dispatch_request
app_1                       |     return self.ensure_sync(self.view_functions[rule.endpoint])(**req.view_args)
app_1                       |   File "/usr/local/lib/python3.9/site-packages/flask_security/decorators.py", line 509, in decorated_view
app_1                       |     return fn(*args, **kwargs)
app_1                       | TypeError: new_address_ep() missing 1 required positional argument: 'address_type'
```

File: templates/lightning/index.html (hunk 2 above)
Comment (eoliveros) [reply templates/lightning/index.html:31]: with the change above theres an error. look at comment from above.

File: templates/lightning/list_transactions.html (hunk 3 above)
Comment (eoliveros) [reply templates/lightning/list_transactions.html:23]: this is to remove the "," at the list of output.

File: ln_wallet_endpoint.py (hunk 1 above)
Comment (eoliveros) [reply ln_wallet_endpoint.py:36]: ahhh got it... making  a push later.

File: templates/lightning/list_transactions.html (hunk 3 above)
Comment (eoliveros) [reply templates/lightning/list_transactions.html:23]: rearranging it to shorten.

**Pull request merged**